spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,642 @@
+    assert_string_fields(domain, %w(parentType name description))
+    assert_integer_fields(domain, %w(backHaulVNID serviceID customerID labelID ECMPCount domainID domainVLANID))
+    assert_boolean_property_fields(domain, %w(stretched globalRoutingEnabled leakingEnabled BGPEnabled))
+    assert_entity_scope(domain['entityScope'])
+    assert_route_property_fields(domain, %w(routeDistinguisher routeTarget backHaulRouteDistinguisher backHaulRouteTarget importRouteTarget exportRouteTarget))
+    assert_enabled_disabled_inherited_fields(domain, %w(maintenanceMode underlayEnabled encryption multicast PATEnabled DPI))
+  end
+
+  def assert_zones(zones)
+    expect(zones.count).to be > 0
+    assert_zone(zones.first)
+  end
+
+  def assert_zone(zone)
+    expect(zone).to include(
+      "children",
+      "parentType",
+      "entityScope",
+      "lastUpdatedBy",
+      "lastUpdatedDate",
+      "creationDate",
+      "address",
+      "netmask",
+      "name",
+      "dynamicIpv6Address",
+      "description",
+      "maintenanceMode",
+      "publicZone",
+      "encryption",
+      "owner",
+      "ID",
+      "parentID",
+      "externalID",
+      "IPv6Address",
+      "IPType",
+      "numberOfHostsInSubnets",
+      "templateID",
+      "policyGroupID",
+      "multicast",
+      "associatedMulticastChannelMapID",
+      "DPI"
+    )
+
+    assert_time_fields(zone, %w(lastUpdatedDate creationDate))
+    assert_identifier_fields(zone, %w(ID lastUpdatedBy owner parentID))
+    assert_string_fields(zone, %w(parentType name))
+    assert_integer_fields(zone, %w(numberOfHostsInSubnets policyGroupID))
+    assert_boolean_property_fields(zone, %w(publicZone))
+    assert_entity_scope(zone['entityScope'])
+    assert_ip_type(zone['IPType'])
+    assert_enabled_disabled_inherited_fields(zone, %w(maintenanceMode encryption multicast DPI))
+  end
+
+  def assert_subnets(subnets)
+    expect(subnets.count).to be > 0
+    assert_subnet(subnets.first)
+  end
+
+  def assert_subnet(subnet)
+    expect(subnet).to include(
+      "children",
+      "parentType",
+      "entityScope",
+      "lastUpdatedBy",
+      "lastUpdatedDate",
+      "creationDate",
+      "address",
+      "netmask",
+      "name",
+      "dynamicIpv6Address",
+      "gateway",
+      "description",
+      "maintenanceMode",
+      "routeDistinguisher",
+      "routeTarget",
+      "vnId",
+      "underlayEnabled",
+      "underlay",
+      "entityState",
+      "splitSubnet",
+      "encryption",
+      "owner",
+      "ID",
+      "parentID",
+      "externalID",
+      "IPv6Address",
+      "IPType",
+      "IPv6Gateway",
+      "serviceID",
+      "gatewayMACAddress",
+      "PATEnabled",
+      "policyGroupID",
+      "public",
+      "templateID",
+      "associatedSharedNetworkResourceID",
+      "DHCPRelayStatus",
+      "proxyARP",
+      "multicast",
+      "associatedMulticastChannelMapID",
+      "DPI",
+      "useGlobalMAC"
+    )
```

Comment:
I just got wondering: do we even need to test for presence explicitly? I mean, we're testing each attribute in more details anyway so we could probably just enrich the "bulk" functions (`assert_time_fields`, `assert_identifier_fields`, ...) with presence check.

This way we wouldn't need to list all the attributes twice.